cranelift/jit/src/compiled_blob.rs

```diff
@@ -93,6 +93,31 @@ impl CompiledBlob {
                     let imm26 = (diff as u32) << chop >> chop;
                     unsafe { modify_inst32(iptr, |inst| inst | imm26) };
                 }
+                Reloc::Aarch64AdrGotPage21 => {
+                    // Set the immediate value of an ADRP to bits [32:12] of X; check that –2^32 <= X < 2^32
+                    let base = get_got_entry(name);
+                    let what = unsafe { base.offset(isize::try_from(addend).unwrap()) };
+                    let what_page = (what as usize) & !0xfff;
+                    let at_page = (at as usize) & !0xfff;
+                    let pcrel = i32::try_from((what_page as isize) - (at_page as isize)).unwrap();
+                    let val = pcrel >> 12;
+
+                    let immlo = ((val as u32) & 0b11) << 29;
+                    let immhi = (((val as u32) >> 2) & &0x7ffff) << 5;
+                    let mask = !((0x7ffff << 5) | (0b11 << 29));
+                    unsafe { modify_inst32(at as *mut u32, |adrp| (adrp & mask) | immlo | immhi) };
+                }
+                Reloc::Aarch64Ld64GotLo12Nc => {
+                    // Set the LD/ST immediate field to bits 11:3 of X. No overflow check; check that X&7 = 0
+                    let base = get_got_entry(name);
+                    let what = unsafe { base.offset(isize::try_from(addend).unwrap()) };
```

Comment:
We only emit `Aarch64Ld64GotLo12Nc` relocations with an `addend` of `0`. Would you mind removing the offset, and instead asserting that it's `0`?